server/app/mutations/grid_certificates/common.rb

```diff
@@ -64,5 +69,84 @@ def validate_dns_record(domain, expected_record)
     rescue
       false
     end
+
+    # @param domain_authorization [GridDomainAuthorization]
+    def validate_domain_authorization(authz)
+
+      case authz.authorization_type
+      when 'dns-01'
+        # Check that the expected DNS record is already in place
+        unless check_dns_record(authz.domain, authz.challenge_opts['record_content'])
+          add_error(:dns_record, :invalid, "Expected DNS record not present for domain #{authz.domain}") # XXX: validations error type
```

Comment:
The mutations error should be related to the input fields... so these should really be like `add_error(:domains, :dns_verification_challenge, "...")`

Reply:
I agree.